src/GNDStk/Node/test/child.test.cpp

```diff
@@ -0,0 +1,150 @@
+
+#include "catch.hpp"
+#include "GNDStk.hpp"
+using namespace njoy::GNDStk;
+
+
+// -----------------------------------------------------------------------------
+// Some child_ts
+// -----------------------------------------------------------------------------
+
+// temperature_t
+struct temperature_t {
+   // just the metadata, for this example
+   double value;
+   std::string unit;
+};
+
+inline void convert(const Node<> &n, temperature_t &temp)
+{
+   temp.value = n(meta::dvalue);
+   temp.unit  = n(meta::unit);
+}
+
+// foo_t, bar_t
+struct foo_t { std::string foo_id; };
+struct bar_t { std::string bar_id; };
+
+inline void convert(const Node<> &n, foo_t &foo)
+   { bool found; foo.foo_id = n(meta::id,found); }
+inline void convert(const Node<> &n, bar_t &bar)
+   { bool found; bar.bar_id = n(meta::id,found); }
+
+// isotope_t
+struct isotope_t {
+   std::string symbol;
+   int A;
+   const Node<> *nuclides;
+};
+
+inline void convert(const Node<> &n, isotope_t &iso)
+{
+   iso.symbol = n(meta::symbol);
+   iso.A = n(meta::A);
+   iso.nuclides = &n(child::nuclides);
+}
+
+// keywords with find::one
+auto temperature =
+   keyword.child<temperature_t,find::one>("temperature");
+auto styles =
+   keyword.child<void,find::one>("styles");
+auto documentations =
+   keyword.child<std::variant<foo_t,bar_t>,find::one>("documentations");
+
+// keywords with find::all
+auto isotope =
+   keyword.child<isotope_t,find::all>("isotope");
+auto isotope_node =
+   keyword.child<void,find::all>("isotope");
+auto nuclide_foo_or_bar_node =
+   keyword.child<std::variant<foo_t,bar_t>,find::all>("nuclide");
+
+
+
+// -----------------------------------------------------------------------------
```

Comment:
I don't see how this adds much—if anything—to what was done in `call.test.cpp`. There are some other variables and such, but I don't see that they are being tested. Am I missing something?

Reply:
It looks like I must have written one of these first, then copied to the other and changed only a few things; there's some amount of repetition in there. In their respective SCENARIO blocks, the call test has more calls to (), while the child test has more calls to .child(). These relatively small differences appear among lots of similarities, so they're relatively hard to spot. Ultimately, the parenthesis operators end up calling a meta() or child() function under the hood, so it stands to reason that tests will (or at least can) be similar in many respects. At some point it would probably be a good idea to review the entire test suite, to ensure that coverage is still good; and perhaps to change things up a bit more between the tests of similar functions, if for no other reason than to avoid apparent repetition.